sky/resources.py

```diff
@@ -944,9 +941,20 @@ def less_demanding_than(self,
             if other.disk_tier is None:
                 return False
             if self.disk_tier != other.disk_tier:
-                types = ['low', 'medium', 'high']
-                return types.index(self.disk_tier) < types.index(
-                    other.disk_tier)
+                if self.cloud is not None:
+                    self_tier = self.cloud.translate_disk_tier(self.disk_tier)
```

Comment:
What does it mean to translate the disk_tier when instance type is not provided? Does it work for Azure?

Reply:
It just picks the best disk tier enabled. For Azure, it will always be `medium` and could fail for certain instance type